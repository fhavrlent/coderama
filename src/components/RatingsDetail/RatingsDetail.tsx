import { Descriptions } from "antd";

import { MovieDetailDTO } from "../../api/useFetchMovieById";

export const RatingsDetail = ({ Ratings }: Pick<MovieDetailDTO, "Ratings">) => (
  <Descriptions title="Ratings">
    {Ratings.map(({ Source, Value }) => (
      <Descriptions.Item key={Source} label={Source}>
        {Value}
      </Descriptions.Item>
    ))}
  </Descriptions>
);
