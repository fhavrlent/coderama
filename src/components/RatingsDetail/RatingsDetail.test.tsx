import { render, screen } from "@testing-library/react";

import { RatingsDetail } from "./RatingsDetail";

describe("RatingDetail", () => {
  it("should render correctly", () => {
    const { container } = render(
      <RatingsDetail
        Ratings={[
          { Source: "Internet Movie Database", Value: "8.9/10" },
          { Source: "Rotten Tomatoes", Value: "92%" },
          { Source: "Metacritic", Value: "80/100" },
        ]}
      />
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText("Internet Movie Database")).toBeInTheDocument();
    expect(screen.getByText("Rotten Tomatoes")).toBeInTheDocument();
  });
});
