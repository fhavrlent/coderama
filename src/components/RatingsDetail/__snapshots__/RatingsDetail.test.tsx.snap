// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingDetail should render correctly 1`] = `
<div>
  <div
    class="ant-descriptions"
  >
    <div
      class="ant-descriptions-header"
    >
      <div
        class="ant-descriptions-title"
      >
        Ratings
      </div>
    </div>
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <td
              class="ant-descriptions-item"
              colspan="1"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Internet Movie Database
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  8.9/10
                </span>
              </div>
            </td>
            <td
              class="ant-descriptions-item"
              colspan="1"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Rotten Tomatoes
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  92%
                </span>
              </div>
            </td>
            <td
              class="ant-descriptions-item"
              colspan="1"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Metacritic
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  80/100
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
